### PR TITLE
Implement upload in Folder

### PR DESCRIPTION
**Pull request by skjnldsv**
https://user-images.githubusercontent.com/14975046/187891817-048e699e-14c6-422d-98b6-21937b639066.mp4

### CODE REVIEW

**Comment by skjnldsv** [comment]
@artonge ready for a first review! If you have any questions :)
Currently only file upload in folders.
I did not use the default folder as if you're displaying the root and upload a file, you would expect this to be in the Root folder I think

File: src/views/Folders.vue [hunk 1]
```diff
@@ -29,14 +29,24 @@
...
-			key="navigation"
-			v-bind="folder"
+	<div v-else-if="!initializing">
+		<HeaderNavigation key="navigation"
+			:loading="loading"
+			:path="folder.filename"
+			:title="folder.basename"
 			:root-title="rootTitle"
-			:show-actions="true" />
+			@refresh="onRefresh">
```
Comment (skjnldsv) [inline src/views/Folders.vue:43]: Clicking the title refreshes the view

File: src/views/Folders.vue [hunk 2]
```diff
@@ -29,14 +29,24 @@
...
+			<UploadPicker :accept="allowedMimes"
+				:destination="folder.filename"
+				:multiple="true"
+				@uploaded="onRefresh" />
```
Comment (skjnldsv) [inline src/views/Folders.vue:47]: Every uploaded file request a refresh. For big folders, that could be expensive, maybe we could fetch the newly uploaded file only? Not sure the albums API allows it :/

Comment (artonge) [reply src/views/Folders.vue:47]: This is to progressively display the files will they are uploaded ?

File: src/components/FolderTagPreview.vue [hunk 3]
```diff
@@ -65,7 +65,7 @@ export default {
 			default: 'icon-folder',
 		},
 		id: {
-			type: Number,
+			type: [Number, String],
```
Comment (marcelklehr) [inline src/components/FolderTagPreview.vue:68]: Whaat, I didn't know that was a thing. Nice!

File: src/views/Folders.vue (hunk 2 above)
Comment (marcelklehr) [reply src/views/Folders.vue:47]: Shouldn't it be enough to add the file id to the folder list without refreshing the full view?

Comment (skjnldsv) [reply src/views/Folders.vue:47]: We don't know the File id without fetching the content :see_no_evil: 

> This is to progressively display the files will they are uploaded ?

This or update the view after upload

Comment (artonge) [reply src/views/Folders.vue:47]: Might be best to do a propfind on the new file, this way we are sure to have all the data, and less expensive than propfinding all the list. This is what files is doing apparently.

**Review by artonge — APPROVED** [review]

**Pull request merged**